cplugin/client/client.go

```diff
@@ -0,0 +1,113 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"errors"
+	"os/exec"
+
+	"github.com/conduitio/conduit-connector-protocol/cplugin"
+	v1 "github.com/conduitio/conduit-connector-protocol/cplugin/v1"              //nolint:staticcheck // v1 is used for backwards compatibility
+	clientv1 "github.com/conduitio/conduit-connector-protocol/cplugin/v1/client" //nolint:staticcheck // v1 is used for backwards compatibility
+	v2 "github.com/conduitio/conduit-connector-protocol/cplugin/v2"
+	clientv2 "github.com/conduitio/conduit-connector-protocol/cplugin/v2/client"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+)
+
+// NewClient creates a new plugin client. The provided context is used to kill
+// the process (by calling os.Process.Kill) if the context becomes done before
+// the plugin completes on its own. Path should point to the plugin executable.
+func NewClient(
```

Comment:
The comment appears to be a bit outdated, since I don't see the mentioned context being used.